ft_scalpcurrentdensity.m

```diff
@@ -182,35 +184,56 @@
   data.trial{trlop} = data.trial{trlop}(dataindx,:);
 end
 
+% remove all junk fields from the electrode array
+tmp  = elec;
+elec = [];
+% select channel positions where potential is known
+goodchan    = setdiff(data.label,cfg.badchannel);
+goodchanidx = ismember(tmp.label, goodchan);
+goodchanpos = tmp.chanpos(goodchanidx,:);
+allchanpos  = tmp.chanpos;
+if isfield(tmp, 'elecpos')
+  elec.elecpos = tmp.elecpos;
+end
+elec.label   = tmp.label;
+
 % compute SCD for each trial
 if strcmp(cfg.method, 'spline')
-
   ft_progress('init', cfg.feedback, 'computing SCD for trial...')
   for trlop=1:Ntrials
-    % do not compute interpolation, but only one value at [0 0 1]
-    % this also gives L1, the laplacian of the original data in which we
-    % are interested here
-
-    ft_progress(trlop/Ntrials, 'computing SCD for trial %d of %d', trlop, Ntrials);
-    [V2, L2, L1] = splint(elec.chanpos, data.trial{trlop}, [0 0 1], cfg.order, cfg.degree, cfg.lambda);
-    scd.trial{trlop} = L1;
+      % do compute interpolation
+      ft_progress(trlop/Ntrials, 'computing SCD for trial %d of %d', trlop, Ntrials);
+      if ~isempty(cfg.badchannel) % if any bad channels to interpolate: compute scd for all channels
+          fprintf('computing scd also at locations of bad channels');
+          [V2, L2, L1] = splint(goodchanpos, data.trial{trlop}(goodchanidx,:), allchanpos, cfg.order, cfg.degree, cfg.lambda);
```

Comment:
goodchanidx is sorted according to the electrode channels, not sorted according to the data channels. They could be different.

Reply:
Yes, you're right! 
Where I previously defined goodchanpos, let's separately define
goodChanElecidx (to retrieve channel positions of valid channels from the elec file) and goodChanDataidx (for selection of valid channels in the data when computing the spline). So from line 186 onwards:

goodChan = setdiff(data.label,cfg.badchannel);
goodChanDataidx = ismember(data.label, goodChan);
goodChanElecidx = ismember(tmp.label, goodChan);
goodchanpos = tmp.chanpos(goodChanElecidx ,:);
allchanpos = tmp.chanpos;
if isfield(tmp, 'elecpos')
  elec.elecpos = tmp.elecpos;
end
elec.label   = tmp.label;

% compute SCD for each trial
if strcmp(cfg.method, 'spline')
  ft_progress('init', cfg.feedback, 'computing SCD for trial...')
  for trlop=1:Ntrials
      % do compute interpolation
      ft_progress(trlop/Ntrials, 'computing SCD for trial %d of %d', trlop, Ntrials);
      if ~isempty(cfg.badchannel) % if any bad channels to interpolate: compute scd for all channels
          [V2, L2, L1] = splint(goodchanpos, data.trial{trlop}(goodChanDataidx,:), allchanpos, cfg.order, cfg.degree, cfg.lambda);
          scd.trial{trlop} = L2;
      else % if all channels good: just compute scd for input channels, specify arbitrary single channel to-be-discarded for interpolation (saves >50% computation time)
          [V2, L2, L1] = splint(goodchanpos, data.trial{trlop}(goodChanDataidx,:), [0 0 1], cfg.order, cfg.degree, cfg.lambda);
          scd.trial{trlop} = L1;
      end
  end
  ft_progress('close');